Add monitor sample

chiranjeevi2776 opened the pull request:
This sample sets up Wi-Fi in monitor mode, parses incoming packets, and displays statistics at regular intervals.
SHEL- 1941

NordicBuilder commented:
## Test specification

### [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-13989/1/)
- Integration Platforms

### [CI/Jenkins/integration](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-13989/1/)
| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| test-sdk-wifi | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-13989/1/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

chiranjeevi2776 commented:
@richabp, can you please review the README.rst. It needs to be incorporated into the 2.6.0 release.

chiranjeevi2776 commented:
@richabp, can you please correct me what could be the wrong in below error log.
/home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrf/src/releases_and_maturity/releases/release-notes-changelog.rst:667: WARNING: term not in glossary: 'Monitor'

I have changed doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst with below

> * Added the :ref:`wifi_monitor_sample` sample that demonstrates how to start a nRF70 Series device in :term:`Monitor` mode.

richabp commented:
> @richabp, can you please correct me what could be the wrong in below error log. /home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrf/src/releases_and_maturity/releases/release-notes-changelog.rst:667: WARNING: term not in glossary: 'Monitor'
> 
> I have changed doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst with below
> 
> > * Added the :ref:`wifi_monitor_sample` sample that demonstrates how to start a nRF70 Series device in :term:`Monitor` mode.

To use the :term: directive, the term "Monitor" should be included in the Glossary here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/glossary.html.

chiranjeevi2776 commented:
> > @richabp, can you please correct me what could be the wrong in below error log. /home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrf/src/releases_and_maturity/releases/release-notes-changelog.rst:667: WARNING: term not in glossary: 'Monitor'
> > I have changed doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst with below
> > > * Added the :ref:`wifi_monitor_sample` sample that demonstrates how to start a nRF70 Series device in :term:`Monitor` mode.
> 
> To use the :term: directive, the term "Monitor" should be included in the Glossary here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/glossary.html.

Updated glossary with Monitor.

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-13989
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

bama-nordic commented on samples/wifi/monitor/README.rst:
```diff
@@ -0,0 +1,110 @@
+.. _wifi_monitor_sample:
+
+Wi-Fi: Monitor
```
The way I see it, this sample should be a raw capture sample which can be configured in monitor mode and (later) promiscuous mode . Let's discuss this internally and modify accordingly.

chiranjeevi2776 replied:
Following internal discussions, maintain "Monitor" as a distinct example, and treat "promiscuous mode" as a separate illustration.

chiranjeevi2776 commented on samples/wifi/monitor/README.rst:
```diff
@@ -0,0 +1,110 @@
+The sample demonstrates how to configure nrf70 Series device in Monitor mode.
+It analyzes the Wi-Fi incoming packets on a raw socket and prints the packet statistics at fixed internal.
+
+* :kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT`: Wait duration for printing Wi-Fi packet statistics.
```
Done.

richabp commented on samples/wifi/monitor/README.rst:
```diff
@@ -0,0 +1,125 @@
+   :local:
+   :depth: 2
+
+The Monitor sample demonstrates how to set monitor mode, analyzing incoming Wi-Fi packets, and printing packet statistics.
```
```suggestion
The Monitor sample demonstrates how to set monitor mode, analyze incoming Wi-Fi® packets, and print packet statistics.
```

richabp replied:
**Wi-Fi®** seems to be missed.

richabp commented on samples/wifi/monitor/README.rst:
```diff
@@ -0,0 +1,125 @@
+Overview
+********
+
+The sample demonstrates how to configure nrf70 Series device in Monitor mode.
```
```suggestion
The sample demonstrates how to configure the nRF70 Series device in Monitor mode.
```

richabp replied:
nRF- **RF** is capitalized.

richabp commented on samples/wifi/monitor/README.rst:
```diff
@@ -0,0 +1,125 @@
+********
+
+The sample demonstrates how to configure nrf70 Series device in Monitor mode.
+It analyzes the Wi-Fi incoming packets on a raw socket and prints the packet statistics at fixed internal.
```
```suggestion
It analyzes the incoming Wi-Fi packets on a raw socket and prints the packet statistics at a fixed interval.
```

richabp commented on samples/wifi/monitor/README.rst:
```diff
@@ -0,0 +1,125 @@
+The sample demonstrates how to configure nrf70 Series device in Monitor mode.
+It analyzes the Wi-Fi incoming packets on a raw socket and prints the packet statistics at fixed internal.
+
+* :kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT`: Wait duration for printing Wi-Fi packet statistics in seconds.
```
```suggestion
To set the wait duration for printing Wi-Fi packet statistics in seconds, use the :kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT` Kconfig option.
```

richabp commented on samples/wifi/monitor/README.rst:
```diff
@@ -0,0 +1,125 @@
+Dependencies
+************
+
+This sample uses the following library:
```
```suggestion
This sample uses the following |NCS| library:
```

richabp commented on samples/wifi/monitor/README.rst:
```diff
@@ -0,0 +1,125 @@
+.. include:: /includes/build_and_run_ns.txt
+
+To build for the nRF7002 DK, use the ``nrf7002dk_nrf5340_cpuapp`` build target.
+The following are examples of the CLI commands:
+
+  .. code-block:: console
+
+     west build -b nrf7002dk_nrf5340_cpuapp
```
```suggestion
The following is an example of the CLI command:

.. code-block:: console

   west build -b nrf7002dk_nrf5340_cpuapp
```

richabp commented on doc/nrf/glossary.rst:
```diff
@@ -442,6 +442,9 @@ Glossary
       An index defined by the Wi-Fi standards where each index specifies a combination of modulation scheme and forward error correction coding rate.
 
+   Monitor
+      A network interface mode that enables the capturing and analysis of all incoming IEEE 802.11 packets.
```
```suggestion
      A network interface mode that enables the capture and analysis of all incoming IEEE 802.11 packets.
```

greg-fer replied:
Isn't this Wi-Fi specific? Would be good to mention this in the term definition, such as:
```suggestion
      A network interface mode for nRF70 Series devices that enables the capture and analysis of all incoming IEEE 802.11 packets.
```

richabp replied:
@chiranjeevi2776 -. Please see this suggestion.

richabp commented on samples/wifi/monitor/README.rst:
```diff
@@ -0,0 +1,126 @@
+The sample demonstrates how to configure the nrf70 Series device in Monitor mode.
+It analyzes the incoming Wi-Fi packets on a raw socket and prints the packet statistics at a fixed interval.
+
+To set the wait duration for printing Wi-Fi packet statistics in seconds, use the
+:kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT` Kconfig option.
```
```suggestion
To set the wait duration for printing Wi-Fi packet statistics in seconds, use the :kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT` Kconfig option.
```

krga2022 commented:
Only control frame counts are updating. Management frames and data frame counts is 0.

`[02:44:37.250,152] <inf> monitor: Management Frames:
[02:44:37.250,183] <inf> monitor:       Beacon Count: 0
[02:44:37.250,183] <inf> monitor:       Probe Request Count: 0
[02:44:37.250,183] <inf> monitor:       Probe Response Count: 0
[02:44:37.250,183] <inf> monitor: Control Frames:
[02:44:37.250,213] <inf> monitor:        Ack Count 17954
[02:44:37.250,213] <inf> monitor:        RTS Count 0
[02:44:37.250,213] <inf> monitor:        CTS Count 0
[02:44:37.250,213] <inf> monitor:        Block Ack Count 50
[02:44:37.250,213] <inf> monitor:        Block Ack Req Count 0
[02:44:37.250,244] <inf> monitor: Data Frames:
[02:44:37.250,244] <inf> monitor:       Data Count: 0
[02:44:37.250,244] <inf> monitor:       QoS Data Count: 0
[02:44:37.250,244] <inf> monitor:       Null Count: 0
[02:44:37.250,274] <inf> monitor:       QoS Null Count: 0`

chiranjeevi2776 commented:
> Only control frame counts are updating. Management frames and data frame counts is 0.
> 
> `[02:44:37.250,152] <inf> monitor: Management Frames: [02:44:37.250,183] <inf> monitor: Beacon Count: 0 [02:44:37.250,183] <inf> monitor: Probe Request Count: 0 [02:44:37.250,183] <inf> monitor: Probe Response Count: 0 [02:44:37.250,183] <inf> monitor: Control Frames: [02:44:37.250,213] <inf> monitor: Ack Count 17954 [02:44:37.250,213] <inf> monitor: RTS Count 0 [02:44:37.250,213] <inf> monitor: CTS Count 0 [02:44:37.250,213] <inf> monitor: Block Ack Count 50 [02:44:37.250,213] <inf> monitor: Block Ack Req Count 0 [02:44:37.250,244] <inf> monitor: Data Frames: [02:44:37.250,244] <inf> monitor: Data Count: 0 [02:44:37.250,244] <inf> monitor: QoS Data Count: 0 [02:44:37.250,244] <inf> monitor: Null Count: 0 [02:44:37.250,274] <inf> monitor: QoS Null Count: 0`

Check with latest ncs main..

richabp commented on samples/wifi/monitor/Kconfig:
```diff
@@ -0,0 +1,25 @@
+	int "Configure the channel for Monitor mode sample"
+	default 1
+	help
+	  Specify the channel for receiving WiFi packets in Monitor mode.
+	  The valid channel range for 2G is 1 to 14
+	  The valid channel range for 5G starts from 36, 40, 44 up to 165.
```
```suggestion
	  Specify the channel for receiving Wi-Fi packets in Monitor mode.
	  The valid channel range for 2G is 1 to 14.
	  The valid channel range for 5G starts from 36, 40, 44, up to 165.
```
